src/app/actions/city-list-actions/actions.tsx

```diff
@@ -0,0 +1,25 @@
+import { CityWeatherData } from "../../contracts/city-weather-data";
+
+export class SubmitFavorite {
+    constructor(private data: CityWeatherData) {
+    }
+    public get Data(): CityWeatherData {
+        return this.data;
+    }
+}
+export class DeleteFavorite {
+    constructor(private city: string) {
+    }
+    public get City(): string {
+        return this.city;
+    }
+}
+export class DeleteAllFavorites { }
+
+export class SubmitSelectedCity {
+    constructor(private city: CityWeatherData) {
```

Comment:
Suggest use more targeted name for variables. `city` -> `cityWeatherData` or something similar (depends on situation).